src/components/Editor/SearchBar.js

```diff
@@ -308,10 +316,19 @@ class SearchBar extends Component {
 
     if (modifiers) {
       const findFnc = rev ? findPrev : findNext;
-      const newIndex = findFnc(ctx, query, true, modifiers.toJS());
+      const findPos = findFnc(ctx, query, true, modifiers.toJS());
+      const { ch: newIndex } = findPos;
+      const matchedLocationIndex = matchedLocations
+        ? matchedLocations.findIndex(elm => {
+            return elm.line == findPos.line && elm.ch == findPos.ch;
+          })
+        : -1;
+
       updateSearchResults({
-        index: newIndex,
-        count
+        matchedLocations,
+        matchedLocationIndex,
+        count: matchedLocations.length,
```

Comment:
could `matchedLocations` be `matches` with a type definition that makes it clear what is stored?